zerver/views/upload.py

```diff
@@ -156,6 +165,34 @@ def serve_local_file_unauthed(request: HttpRequest, token: str, filename: str) -
     return serve_local(request, path_id, url_only=False)
 
 
+def serve_local_avatar_unauthed(request: HttpRequest, path: str) -> HttpResponseBase:
+    if settings.LOCAL_UPLOADS_DIR is None:
+        # We do not expect clients to hit this URL when using the S3
+        # backend; however, there is no reason to not serve the
+        # redirect to S3 where the content lives.
+        return redirect(
+            get_public_upload_root_url() + path + "?" + request.GET.urlencode(), permanent=True
+        )
+
+    assert settings.LOCAL_AVATARS_DIR is not None
+    local_path = os.path.join(settings.LOCAL_AVATARS_DIR, path)
+    assert_is_local_storage_path("avatars", local_path)
+    if not os.path.isfile(local_path):
+        return HttpResponseNotFound("<p>File not found</p>")
+
+    if settings.DEVELOPMENT:
+        response: HttpResponseBase = FileResponse(open(local_path, "rb"), as_attachment=False)
+        guessed_type = guess_type(local_path)[0]
+        if guessed_type is not None:
+            response["Content-Type"] = guessed_type
+    else:
+        response = HttpResponse()
+        response["X-Accel-Redirect"] = quote(f"/internal/user_avatars/{path}")
+        del response["Content-Type"]
```

Comment:
Can you copy the explanation for why we don't set caching headers for avatars from the commit message to a comment inside this function? I think that's a fact that will be very helpful for future readers comparing this code path to the uploads version.

(I suppose it's possible we could even make this use the same function, with just an extra parameter, but I'm not sure that'd be cleaner)

Reply:
Sure, done.